### DISCUSSION TITLE
IKEv2 + SCEP Cert VPN configurations via Intune to Android

### DISCUSSION THREAD

**Discussion by Stuart-McGuire** [strongswan/strongswan#2208] [Q&A]
Now that Android configs can be pushed via Intune, I was wondering if it was possible to create a profile which either:
a) Auto chooses the user cert
b) Allows the user to pick the user cert.

We have implemented SCEPMAN which pushes certs out to the Android devices fine. Each user's cert has a different name (the first part of their email address). This works well if a user creates a VPN from scratch and puts in all the connection details and chooses their cert. 

My problem is
 a) that each cert has a different name so I'm not sure what to put in the JSON config for this
 b) Allow users to pick a cert. Currently they get the message 'This VPN profile is managed by your administrator and can't be modified'.

Any help would be greatly appreciated.

**Comment by tobiasbrunner** [comment]
Currently, the managed profile is intended to provide the user certificate/key, which is automatically installed on the system. That assignment can't be changed later by the user.

> a) that each cert has a different name so I'm not sure what to put in the JSON config for this

What do you mean? What name? What JSON config?

> b) Allow users to pick a cert. Currently they get the message 'This VPN profile is managed by your administrator and can't be modified'.

Since passwords can be changed in the GUI, we could perhaps consider allowing users to select a different client certificate (or one at all if none was installed with the profile). But not sure how difficult that would be to implement considering that the certificates are currently fully managed.

**Reply by Stuart-McGuire** [reply]
Thank you for taking the time to reply. 

> Currently, the managed profile is intended to provide the user certificate/key, which is automatically installed on the system. That assignment can't be changed later by the user.

Ah, ok - I thought the managed profile linked to the cert which was already one the device, rather than provided it. I guess this means everyone shares the same cert?

> What do you mean? What name? What JSON config?

The JSON config that is edited in the App Configuration Policy within Intune ( I assume this IS the managed profile then).

>Since passwords can be changed in the GUI, we could perhaps consider allowing users to select a different client certificate (or one at all if none was installed with the profile). But not sure how difficult that would be to implement considering that the certificates are currently fully managed.

Ok - one to keep an eye out for then. :)

**Reply by tobiasbrunner** [reply]
> Ah, ok - I thought the managed profile linked to the cert which was already one the device, rather than provided it. I guess this means everyone shares the same cert?

Why? Can't your management tool distribute individualized configs for each user?

> The JSON config that is edited in the App Configuration Policy within Intune ( I assume this IS the managed profile then).

Ah, no idea how Intune works, so can't help you with that.

**Comment by maluueu** [comment]
@tobiasbrunner no, no EMM can deploy individualized configs which include the cert. The device/user certs are requested by the device using SCEP or some other mechanism. The EMM should never be able to see certificate/key of the device/user cert.
The only thing the EMM can include in the config is a reference to the on-device trust-store to let you know which certificate should be used. If this is not possible to do atm, the EMM integration seems to be broken/or at least missing support for EMM deployed device/user certs.

This is also not specific only to Microsoft Intune

**Reply by tobiasbrunner** [reply]
> no, no EMM can deploy individualized configs which include the cert.

I don't think that's true. The company that provided the original patches clearly is able to do that in their EMM solution (they ship the PKCS#12 containers and the corresponding passwords via managed configuration, which allow the app to import the key/certificate into the KeyChain).


> The only thing the EMM can include in the config is a reference to the on-device trust-store to let you know which certificate should be used.

That might be something to add in the future (i.e. allow configuring just an alias for the KeyChain).

**Reply by ghost** [reply]
> > no, no EMM can deploy individualized configs which include the cert.
> 
> I don't think that's true. The company that provided the original patches clearly is able to do that in their EMM solution (they ship the PKCS#12 containers and the corresponding passwords via managed configuration, which allow the app to import the key/certificate into the KeyChain).

Hello everyone & @tobiasbrunner ! Other EMM but same issue:
- One certificate per android device
- Certificate already installed on the device
- I can't put the per android device PKCS#12 cert on the managed configurations of the EMM

Impossible to select the certificate with the managed VPN profile :
![Screenshot_20241029_170728_strongSwan VPN Client](https://github.com/user-attachments/assets/f6b46759-eeb0-409f-b372-a8559d3b11f3)

No problem with a manual profile :
![Screenshot_20241029_170807_Key Chain](https://github.com/user-attachments/assets/17efc280-c124-4e74-b416-aea571f7400e)

Maybe if it's possible, let the user choose the cert or config in managed configurations the cert name.

Thanks for your help and have a good day

**Reply by MartinMDS** [reply]
Hello everyone ! Other EMM too but same issue, I'm stucked
Thanks @arthdzn for the screenshots

**Reply by ludo4prez** [reply]
Hi everyone, I've got the same issue as @arthdzn and look forward for your answer.

**Reply by tobiasbrunner** [reply]
Is this a limitation of these EMMs (not being able to assign values to individual devices/users) or a policy issue (having to create the private key on the device)? What EMMs are you guys actually using?

As mentioned before, it might be possible to make the alias configurable via managed config (or maybe make it selectable on the client). But this has currently not much priority.

**Reply by ghost** [reply]
It's mainly the assignment of a unique certificate per device in a managed configuration variable.
I can create a cert + priv key on the device with the EMM but install on the Android cert store.
Intune or Knox (samsung).

Oh too bad...

**Reply by Weatherlights** [reply]
> It's mainly the assignment of a unique certificate per device in a managed configuration variable. I can create a cert + priv key on the device with the EMM but install on the Android cert store. Intune or Knox (samsung).
> 
> Oh too bad...

Same issue Here that makes secure enterprise use of the client not possible. Using Knox and Intune aswell. Will now Check OpenVPN instead.

**Comment by MartinMDS** [comment]
[like]  Martin Deschamps reacted to your message:
________________________________

**Comment by tobiasbrunner** [comment]
As discussed in #2888, with version 2.6.0 of the app, it's possible to distribute the alias of an on-device certificate/key with a managed profile. The MDM has to grant permission to access that alias to the app. Alternatively, if neither a certificate nor an alias is distributed, it's now also possible to manually select a (user-selectable) local certificate when editing a managed profile.
